dbos/dbos.py

```diff
@@ -97,37 +97,70 @@ def __call__(*args: P.args, **kwargs: P.kwargs) -> R: ...
     "READ COMMITTED",
 ]
 
+_dbos_global_instance: Optional[DBOS] = None
+
+
+def get_dbos_instance() -> DBOS:
+    # Currently get / init the global singleton
+    #   (It could also check context?)
+    return DBOS()
+
 
 class DBOS:
+    ### Lifecycles ###
+    # We provide a singleton, created / accessed as `DBOS(args)`
+    #  Access to the the singleton via `DBOS.<thing>`
+    #
+    # If an application wants to control lifecycle of DBOS via singleton:
+    #  Create DBOS with `DBOS()`
+    #   Use DBOS or the instance returned from DBOS()
+    #  clear_global_instance() to get rid of it so that DBOS() returns a new one
+
+    def __new__(
+        cls: Type[DBOS],
+        fastapi: Optional["FastAPI"] = None,
+        config: Optional[ConfigFile] = None,
+    ) -> DBOS:
+        global _dbos_global_instance
+        if _dbos_global_instance is None:
+            _dbos_global_instance = super().__new__(cls)
+            _dbos_global_instance.__init__(fastapi=fastapi, config=config)  # type: ignore
+        return _dbos_global_instance
+
+    @classmethod
+    def clear_global_instance(cls) -> None:
+        global _dbos_global_instance
+        _dbos_global_instance = None
+
     def __init__(
         self, fastapi: Optional["FastAPI"] = None, config: Optional[ConfigFile] = None
     ) -> None:
+        if hasattr(self, "_initialized") and self._initialized:
+            return
+
+        self._initialized: bool = True
```

Comment:
But this is called the first time a decorator is loaded, right?  So if I load a file that contains a decorator, then I pass in a fastapi object, it just ignores the fastapi object?

Reply:
Fixed now.